pyanaconda/payload/dnf/payload.py

```diff
@@ -141,7 +141,11 @@ def set_from_opts(self, opts):
         """
         # Set the source based on opts.method if it isn't already set
         # - opts.method is currently set by command line/boot options
-        if opts.method and (not self.proxy.Sources or self.source_type == DNF_DEFAULT_SOURCE_TYPE):
+
+        # in some cases the CDN is a valid DNF default source
+        default_sources = (DNF_DEFAULT_SOURCE_TYPE, SOURCE_TYPE_CDN)
+        default_source_set = self.source_type in default_sources
```

Comment:
This should be solved differently. CDN shouldn't be a special case. Also, you want to check `opts.method` before working with DBus sources, because `self.proxy` creates a default one.

Reply:
You should remove the constant `DNF_DEFAULT_SOURCE_TYPE` from the code and replace it with `conf.payload.default_source`, so you can drop these ugly workarounds.

Reply:
Used the config option, much cleaner now.